index.js

```diff
@@ -170,6 +173,11 @@ Object.defineProperty(File.prototype, 'contents', {
     if (!isBuffer(val) && !isStream(val) && !isNull(val)) {
       throw new Error('File.contents can only be a Buffer, a Stream, or null.');
     }
+
+    if (isStream(val)) {
+      val = cloneable(val);
+    }
```

Comment:
Are you sure you need this? I would avoid, given you are already wrapping on demand https://github.com/gulpjs/vinyl/pull/99/files#diff-168726dbe96b3ce427e7fedce31bb0bcR85.